src/components/Calls/CallNotification.js

```diff
@@ -0,0 +1,78 @@
+import { h } from 'preact';
+import { useState } from 'preact/compat';
+
+import { Livechat } from '../../api';
+import I18n from '../../i18n';
+import PhoneAccept from '../../icons/phone.svg';
+import PhoneDecline from '../../icons/phoneOff.svg';
+import constants from '../../lib/constants';
+import { Avatar } from '../Avatar';
+import { Button } from '../Button';
+import { createClassName, getAvatarUrl } from '../helpers';
+import { CallIframe } from './CallIFrame';
+import styles from './styles.scss';
+
+
+export const CallNotification = ({ callProvider, callerUsername, url, dispatch, rid } = { callProvider: undefined, callerUsername: undefined, dispatch: undefined, rid: undefined }) => {
+	const [show, setShow] = useState(!!callProvider && !!callerUsername && !!dispatch && !!url);
+	const [isIframe, setIframe] = useState(false);
+	const { token } = Livechat.credentials;
+	const msg = 'JOIN_CALL_MSG';
+
+
+	const acceptClick = async () => {
+		setShow(!{ show });
+		await Livechat.sendMessage({ token, rid, msg });
+		switch (callProvider) {
+			case constants.jitsiCallStartedMessageType: {
+				window.open(url);
+				break;
+			}
+			case constants.webrtcCallStartedMessageType: {
+				// TODO: add webrtc code here
+				setIframe(true);
+				break;
+			}
+		}
+	};
+
+	const declineClick = async () => {
+		await dispatch({ incomingCallAlert: null });
+	};
+
+	const timeout = () => {
+		setTimeout(async function() { setShow(!{ show }); }, 20000);
+	};
+
+	return (
+		<div>
+			{ show
+				? (
+					<div className={createClassName(styles, 'call-notification')}>
+						<div className={createClassName(styles, 'call-notification__content')}>
+							<div className={createClassName(styles, 'call-notification__content-avatar')}>
+								<Avatar
+									src={getAvatarUrl(callerUsername)}
+									large
+								/>
+							</div>
+							<div className={createClassName(styles, 'call-notification__content-message')}>
+								{ I18n.t('Incoming video Call') }
+							</div>
+							<div className={createClassName(styles, 'call-notification__content-actions')}>
+								<Button onClick={declineClick} className={createClassName(styles, 'call-notification__content-actions-decline')}>
+									<PhoneDecline width={20} height={20} /> <span style='margin-left:5px'> {I18n.t('Decline')} </span>
+								</Button>
+								<Button onClick={acceptClick} className={createClassName(styles, 'call-notification__content-actions-accept')} >
+									<PhoneAccept width={20} height={20} /><span style='margin-left:5px'> {I18n.t('Accept')} </span>
+								</Button>
+							</div>
+						</div>
+					</div>
+				)
+				: null
+			}
+		    {timeout()}
+			{ isIframe ? <CallIframe rid={rid} /> : null}
```

Comment:
Not a good practice to add this Iframe Component inside the `CallNotification`component because both of them are not related. This component should be rendered just like the `CallNotification`component is being rendered [here](https://github.com/RocketChat/Rocket.Chat.Livechat/pull/618/files#diff-68fca7346dd568c2624a0a156ad3ba4b5c82359d4dd691c7d6d51cfc68a239beR150) so we need another variable in the `Store` or perhaps we can use [this](https://github.com/RocketChat/Rocket.Chat.Livechat/pull/618/files#diff-68cfdabf9fa2ce99fa19fdbe25785f2622056c22f88b1f846d3ac7ffffea0744R38) one which I'd created